expenses-manager-backend/src/main/java/com/endava/expensesmanager/service/impl/ExpenseServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.endava.expensesmanager.service.impl;
+
+import com.endava.expensesmanager.dto.ExpenseDto;
+import com.endava.expensesmanager.entity.Expense;
+import com.endava.expensesmanager.mapper.ExpenseMapper;
+import com.endava.expensesmanager.repository.ExpenseRepository;
+import com.endava.expensesmanager.service.ExpenseService;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ExpenseServiceImpl implements ExpenseService {
+
+    private final ExpenseRepository expenseRepository;
+    private final ExpenseMapper expenseMapper;
+
+    public ExpenseServiceImpl(ExpenseRepository expenseRepository) {
+        this.expenseRepository = expenseRepository;
+        this.expenseMapper = ExpenseMapper.INSTANCE;
```

Comment:
For mapper I would prefer initialization at declaration time. Just a preference 😄